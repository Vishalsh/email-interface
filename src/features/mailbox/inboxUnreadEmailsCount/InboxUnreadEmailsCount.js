import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Badge from 'components/badge/Badge';
import inboxUnreadEmailsCountSelector from './inboxUnreadEmailsCountSelector';
import classes from './InboxUnreadEmailsCount.module.scss';

export const InboxUnreadEmailsCount = (props) => {
  const {
    readEmailsCount,
    unreadEmailsCount,
    showReadMailsCount,
    showBadge
  } = props;

  const label = showReadMailsCount ? `${unreadEmailsCount}/${readEmailsCount}` : unreadEmailsCount;

  return (
    unreadEmailsCount !== 0 &&
    <>
    {
      showBadge ?
        <Badge className={classes.inboxUnreadEmailsCount} label={label}/>
        :
        `(${label})`
    }
    </>
  );
};

InboxUnreadEmailsCount.defaultProps = {
  showReadMailsCount: false,
  showBadge: true
};

InboxUnreadEmailsCount.propTypes = {
  readEmailsCount: PropTypes.number.isRequired,
  unreadEmailsCount: PropTypes.number.isRequired,
  showReadMailsCount: PropTypes.bool,
  showBadge: PropTypes.bool
};

const mapStateToProps = (store) => ({
  readEmailsCount: store.mailbox.inbox ? store.mailbox.inbox.emails.length : 0,
  unreadEmailsCount: inboxUnreadEmailsCountSelector(store),
});

export default connect(mapStateToProps)(InboxUnreadEmailsCount);
