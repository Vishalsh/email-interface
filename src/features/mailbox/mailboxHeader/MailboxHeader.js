import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/button/Button';
import SearchInput from 'components/searchInput/SearchInput';
import InboxUnreadEmailsCount from 'features/mailbox/inboxUnreadEmailsCount/InboxUnreadEmailsCount';
import { INBOX } from 'constants/mailbox';
import classes from './MailboxHeader.module.scss';

const MailboxHeader = (props) => {
  const {
    mailbox,
    onClickDeleteEmails
  } = props;

  return (
    <header className={classes.header}>
      <div className="row">
        <h1 className={`${classes.name} col-sm-7 col-md-8`}>
          {mailbox}
          {
            mailbox === INBOX &&
            <><InboxUnreadEmailsCount showBadge={false}/></>
          }
        </h1>
        <div className="col-sm-5 col-md-4 end-sm">
          <SearchInput />
        </div>
      </div>
      <div className={`${classes.mailboxActions} row`}>
        <div className="col-sm-8 flex">
          <Button type="default" className={classes.actionButton}>
            <i className={`icon icon-refresh ${classes.iconRefresh}`}/>
            <span>Refresh</span>
          </Button>
          <Button type="default" className={classes.actionButton}>
            <i className="icon icon-eye"/>
          </Button>
          <Button type="default" className={classes.actionButton}>
            <i className="icon icon-bang"/>
          </Button>
          <Button type="default" onClick={onClickDeleteEmails} className={classes.actionButton}>
            <i className="icon icon-delete"/>
          </Button>
        </div>
        <div className="col-sm-4 end-xs flex">
          <Button type="default" className={classes.actionButton}>
            <i className="icon icon-left"/>
          </Button>
          <Button type="default" className={`${classes.actionButton} ${classes.paginationRightButton}`}>
            <i className="icon icon-right"/>
          </Button>
        </div>
      </div>
    </header>
  );
};

MailboxHeader.propTypes = {
  mailbox: PropTypes.string.isRequired,
  onClickDeleteEmails: PropTypes.func.isRequired
};

export default MailboxHeader;
