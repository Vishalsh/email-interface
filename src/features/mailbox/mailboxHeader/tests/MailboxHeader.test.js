import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import InboxUnreadEmailsCount from 'features/mailbox/inboxUnreadEmailsCount/InboxUnreadEmailsCount';
import MailboxHeader from '../MailboxHeader';

describe('MailboxHeader', () => {
  const props = {
    mailbox: 'inbox',
    onClickDeleteEmails: jest.fn(),
  };
  const component = shallow(<MailboxHeader {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should unread emails count only for Inbox', () => {
    expect(component.find(InboxUnreadEmailsCount)).toHaveLength(1);

    const propWithSentMailbox = {
      ...props,
      mailbox: 'sent'
    };
    const componentWithSentMailbox = shallow(<MailboxHeader {...propWithSentMailbox} />);

    expect(componentWithSentMailbox.find(InboxUnreadEmailsCount)).toHaveLength(0);
  });
});