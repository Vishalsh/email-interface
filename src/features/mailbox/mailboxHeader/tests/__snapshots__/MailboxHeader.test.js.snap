// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailboxHeader should render the component 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <h1
      className="name col-sm-7 col-md-8"
    >
      inbox
      <Connect(InboxUnreadEmailsCount)
        showBadge={false}
      />
    </h1>
    <div
      className="col-sm-5 col-md-4 end-sm"
    >
      <SearchInput />
    </div>
  </div>
  <div
    className="mailboxActions row"
  >
    <div
      className="col-sm-8 flex"
    >
      <Button
        className="actionButton"
        disabled={false}
        type="default"
      >
        <i
          className="icon icon-refresh iconRefresh"
        />
        <span>
          Refresh
        </span>
      </Button>
      <Button
        className="actionButton"
        disabled={false}
        type="default"
      >
        <i
          className="icon icon-eye"
        />
      </Button>
      <Button
        className="actionButton"
        disabled={false}
        type="default"
      >
        <i
          className="icon icon-bang"
        />
      </Button>
      <Button
        className="actionButton"
        disabled={false}
        onClick={[MockFunction]}
        type="default"
      >
        <i
          className="icon icon-delete"
        />
      </Button>
    </div>
    <div
      className="col-sm-4 end-xs flex"
    >
      <Button
        className="actionButton"
        disabled={false}
        type="default"
      >
        <i
          className="icon icon-left"
        />
      </Button>
      <Button
        className="actionButton paginationRightButton"
        disabled={false}
        type="default"
      >
        <i
          className="icon icon-right"
        />
      </Button>
    </div>
  </div>
</header>
`;
